modules/services/fb_streetview_ai_suggestions.js

```diff
@@ -0,0 +1,499 @@
+            id: uid,
+            suggestionId: obj.getAttribute("suggestion-id"),
+            visible: getVisible(attrs),
+            tags: getTags(obj),
+            nodes: getNodes(obj),
+        });
+    },
+
+    relation: function relationData(obj, uid) {
+        var attrs = obj.attributes;
+        return new osmRelation({
+            id: uid,
+            suggestionId: obj.getAttribute("suggestion-id"),
+            visible: getVisible(attrs),
+            tags: getTags(obj),
+            members: getMembers(obj)
+        });
+    },
+};
+
+
+// parse <osm-road> element inside <sidewalk-suggestion>
+function praseOsmRoadMeta(xmlEle) {
+    var osmRoadMeta = {};
+    osmRoadMeta.wid = osmEntity.id.fromOSM("way", xmlEle.getAttribute("way-id"));
+    osmRoadMeta.version = xmlEle.getAttribute("version");
+    osmRoadMeta.sidewalk_tag = xmlEle.getAttribute("sidewalk-tag");
+
+    var nodeEles = xmlEle.getElementsByTagName("nd");
+    var nodes = new Array(nodeEles.length);
+    for (var i = 0; i < nodeEles.length; i++) {
+        var n = nodeEles[i];
+        nodes[i] = {
+            nid: osmEntity.id.fromOSM("node", n.getAttribute("ref")),
+            version: n.getAttribute("version"),
+            lat: parseFloat(n.getAttribute("lat")),
+            lon: parseFloat(n.getAttribute("lon"))
+        };
+    }
+    osmRoadMeta.nodes = nodes;
+    return osmRoadMeta;
+}
+
+
+// parse <steet-view-image-set> element inside <sidewalk-suggestion>
+function praseStreetViewImageSet(xmlEle) {
+    var streetViewImageSet = {};
+    streetViewImageSet.id = xmlEle.getAttribute("id");
+    streetViewImageSet.cameraPointingDirection = xmlEle.getAttribute("camera-pointing-direction");
+    streetViewImageSet.relativeSideToRoad = xmlEle.getAttribute("relative-side-to-road");
+
+    var imageEles = xmlEle.getElementsByTagName("street-view-image");
+    var images = new Array(imageEles.length);
+    for (var i = 0; i < imageEles.length; i++) {
+        var img = imageEles[i];
+        images[i] = {
+            key: img.getAttribute("key"),
+            url: img.getAttribute("url"),
+            sidewalkSide: img.getAttribute("sidewalk-side"),
+            lat: parseFloat(img.getAttribute("lat")),
+            lon: parseFloat(img.getAttribute('lon'))
+        };
+    }
+    streetViewImageSet.images = images;
+    return streetViewImageSet;
+}
+
+
+// parse <sidewalk-suggestion>
+function praseSidewalkSuggestion(xmlEle, cubitorContext) {
+    var id = xmlEle.getAttribute("id");
+    if (cubitorContext[id]) return;
+
+    var suggestion = {id: id};
+
+    // we'll support other feature types like crosswalk and speed limit in the future
+    suggestion.featureType = "sidewalk";
+
+    // whether it's suggestion for geometry or tag change
+    suggestion.suggestionType = xmlEle.getAttribute("type");
+
+    xmlEle.childNodes.forEach(function (ele) {
+        if (ele.nodeName === "osm-road") {
+            suggestion.osmRoadMeta = praseOsmRoadMeta(ele);
```

Comment:
This works for how we've defined the dummy data but I can see in the future that we may have multiple OSM ways for a single suggestion. We may want this to be an ordered dict or something of the sort.  Or maybe we can leave this part as is and change the function to create an array similar to how we create a set for the images (I think retaining order could be important for the ways since the roads are ordinal themselves).